### PR TITLE
Fix MADGRAD adding eps to the learning rate

### PR DESCRIPTION
**Pull request by winklemad**
### Problem

MADGRAD computes its step size from `lr + eps`:

```python
lr: float = group['lr'] + eps
_lambda = lr * math.pow(self.state['k'] + 1, 0.5)
```

`eps` is a denominator stabilizer (used in `grad_sum_sq.pow(1/3).add_(eps)`) and should not be part of the learning rate. Because it is, the effective lr is inflated — negligibly for normal lr, but badly for small ones (the fine-tuning / decayed-schedule regime):

| lr | effective (eps=1e-6) | inflation |
|----|----------------------|-----------|
| 1e-3 | 1.001e-3 | ~1x |
| 1e-6 | 2e-6 | **2x** |
| 1e-7 | 1.1e-6 | **11x** |

A step with `lr=0` also still moves the parameter. The reference MADGRAD uses `lamb = lr * sqrt(k + 1)` (no eps on lr).

### Fix

Use `lr` directly; `eps` still stabilizes the cube-root denominator. Added `test_madgrad_learning_rate_not_inflated_by_eps` (lr=0 is now a no-op); `ruff` clean, existing MADGRAD tests pass.

### CODE REVIEW

File: pytorch_optimizer/optimizer/madgrad.py [hunk 1]
```diff
@@ -100,7 +100,7 @@ def step(self, closure: Closure = None) -> Loss:
                 self.init_group(group)
 
             weight_decay, momentum, eps = group['weight_decay'], group['momentum'], group['eps']
-            lr: float = group['lr'] + eps
```
Comment (kozistr) [inline pytorch_optimizer/optimizer/madgrad.py:103]: actually, i followed the original MADGRAD implementation, which adds `eps` to `lr` (https://github.com/facebookresearch/madgrad/blob/main/madgrad/madgrad.py#L101).

And, yes you're right. when we are using lr smaller than eps (default 1e-6), the effective `lr` can become much larger than the original value. The author recommends setting `eps=0` for very small gradient case. (https://github.com/facebookresearch/madgrad/blob/main/madgrad/madgrad.py#L49C13-L49C95)

Secondly, for `lr=0` problem, we can only add `eps` when `lr` is not 0.0 like this! (https://github.com/facebookresearch/madgrad/blob/main/madgrad/madgrad.py#L100)

In conclusion, I think as-is behavior is correct, and we can add `eps` for `lr>0` case!

**Comment by winklemad** [comment]
Good point — you're right that folding eps into `lr` matches the reference MADGRAD, so I've kept that for `lr > 0` and only guarded the `lr == 0` case, so a zero learning rate no longer takes an eps-sized step. Updated the test to assert just that behavior. Thanks!
